### PR TITLE
Add binary distributions for doubles and integers

### PR DESCRIPTION
**Pull request by gonuke**
This adds a new random distribution that samples between two options with a single probability.

### CODE REVIEW

**Review by nuclearkatie — APPROVED** [review]
Thanks for helping with this @gonuke!

**Comment by nuclearkatie** [comment]
Ah, does this need a rebase since I merged #1825 ?

File: src/random_number_generator.h [hunk 1]
```diff
@@ -176,6 +178,26 @@ class ExpontentialDoubleDist : public DoubleDistribution {
     virtual double lambda() { return lambda_; }
 };
 
+
+/// Binary distribution requires twp options and a probability
```
Comment (gonuke) [inline src/random_number_generator.h:182]:
```suggestion
/// Binary distribution requires two options and a probability
```

File: src/random_number_generator.h [hunk 2]
```diff
@@ -306,6 +328,26 @@ class ExponentialIntDist : public IntDistribution {
     virtual double lambda() { return lambda_; }
 };
 
+/// Binary distribution requires twp options and a probability
```
Comment (gonuke) [inline src/random_number_generator.h:331]:
```suggestion
/// Binary distribution requires two options and a probability
```

**Review by gonuke — COMMENTED** [review]
Found 2 typos

**Comment by nuclearkatie** [comment]
Still needs the rebase I think

**Comment by gonuke** [comment]
reabase done

**Pull request merged**